### PR TITLE
Support `wrangler pages secret put|delete|list|bulk`

### LINKED ISSUE #5180: 🚀 Feature Request: Wrangler secret for pages
I want to add secrets to my Cloudflare Page through wrangler (in CI), but I figured out that it's only possible to set secrets for workers (https://developers.cloudflare.com/workers/wrangler/commands/#secret) and not for pages. Any workaround ?

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

This PR supports adding secrets to a Pages project using the Wrangler CLI. Similarly to `wrangler secret`, this PR adds a `wrangler pages secret` set of commands. There are several differences to the Workers commands:
- `--project` is used to specify the project to target, compared to `--name` for Workers.
- The only environments that are supported are `--env=production|preview`
- `secret:bulk` is renamed to `secret bulk`
- The output is slightly friendlier

Fixes #5180

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [x] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [ ] Not necessary because:

### CODE REVIEW

**Review by RamIdeas — APPROVED** [review]

File: packages/wrangler/src/pages/secret/index.ts [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+	}
+	return { env, project, accountId, config };
+}
+
+export const secret = (secretYargs: CommonYargsArgv, subHelp: SubHelp) => {
+	return secretYargs
+		.command(subHelp)
+		.command(
+			"put <key>",
+			"Create or update a secret variable for a Worker",
```
Comment (admah) [inline packages/wrangler/src/pages/secret/index.ts:154]:
```suggestion
			"Create or update a secret variable for a Pages project",
```

File: packages/wrangler/src/pages/secret/index.ts [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+			(yargs) => {
+				return yargs
+					.positional("key", {
+						describe: "The variable name to be accessible in the Worker",
```
Comment (admah) [inline packages/wrangler/src/pages/secret/index.ts:158]:
```suggestion
						describe: "The variable name to be accessible in the Pages project",
```

File: packages/wrangler/src/pages/secret/index.ts [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+					logger.log(`✨ 0 secrets successfully uploaded`);
+					throw new FatalError(
+						`🚨 ${Object.keys(upsertBindings).length} secrets failed to upload`
+					);
+				}
+			}
+		)
+		.command(
+			"delete <key>",
+			"Delete a secret variable from a Worker",
```
Comment (admah) [inline packages/wrangler/src/pages/secret/index.ts:312]:
```suggestion
			"Delete a secret variable from a Pages project",
```

File: packages/wrangler/src/pages/secret/index.ts [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+			async (yargs) => {
+				return yargs
+					.positional("key", {
+						describe: "The variable name to be accessible in the Worker",
```
Comment (admah) [inline packages/wrangler/src/pages/secret/index.ts:316]:
```suggestion
						describe: "The variable name to be accessible in the Pages project",
```

File: packages/wrangler/src/pages/secret/index.ts [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+					await metrics.sendMetricsEvent("delete pages encrypted variable", {
+						sendMetrics: config?.send_metrics,
+					});
+					logger.log(`✨ Success! Deleted secret ${args.key}`);
+				}
+			}
+		)
+		.command(
+			"list",
+			"List all secrets for a Worker",
```
Comment (admah) [inline packages/wrangler/src/pages/secret/index.ts:367]:
```suggestion
			"List all secrets for a Pages project",
```

**Review by admah — CHANGES_REQUESTED** [review]
Left some suggestions for messaging updates.

**Review by admah — APPROVED** [review]
LGTM!

**Pull request merged**